src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -142,6 +143,53 @@ private void ParseCondition(XElement condition, UrlRewriteRuleBuilder builder, P
             }
         }
 
+        private void ParseServerVariables(XElement serverVariables, UrlRewriteRuleBuilder builder)
+        {
+            if (serverVariables == null)
+            {
+                return;
+            }
+
+            foreach (var serverVariable in serverVariables.Elements(RewriteTags.Set))
+            {
+                ParseServerVariable(serverVariable, builder);
+            }
+        }
+
+        private void ParseServerVariable(XElement serverVariable, UrlRewriteRuleBuilder builder)
+        {
+            try
+            {
+                string prefix;
+                ServerVariableType type;
+                var name = serverVariable.Attribute(RewriteTags.Name).Value;
+
+                if (name.StartsWith(ServerVariableConstants.RequestHeaderPrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
ok, not sure if https://github.com/aspnet/BasicMiddleware/pull/184/commits/ded8bbc036133a0a29a48928c780b7597016d3ce is what you had in mind. let me know what you think.